xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java

```diff
@@ -101,23 +102,32 @@ protected Map<Object, ChildLbState> createChildLbMap(ResolvedAddresses resolvedA
    */
   @Override
   public Status acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-    try {
-      resolvingAddresses = true;
-
-      // process resolvedAddresses to update children
-      AcceptResolvedAddrRetVal acceptRetVal =
-          acceptResolvedAddressesInternal(resolvedAddresses);
-      if (!acceptRetVal.status.isOk()) {
-        return acceptRetVal.status;
+    if (lastResolvedAddresses != null) {
+      // Handle deactivated children
+      ClusterManagerConfig config = (ClusterManagerConfig)
+          resolvedAddresses.getLoadBalancingPolicyConfig();
+      ClusterManagerConfig lastConfig = (ClusterManagerConfig)
+          lastResolvedAddresses.getLoadBalancingPolicyConfig();
+      Map<String, PolicySelection> adjChildPolicies = new HashMap<>(config.childPolicies);
+      for (Entry<String, PolicySelection> entry : lastConfig.childPolicies.entrySet()) {
+        ClusterManagerLbState state = (ClusterManagerLbState) getChildLbState(entry.getKey());
+        if (adjChildPolicies.containsKey(entry.getKey())) {
+          if (state.deletionTimer != null) {
+            state.reactivateChild();
```

Comment:
How does `getChildLbState` return a deactivated child when `MultiChildLoadBalancer.getRemovedChildren` now does `ChildLbState childLbState = childLbStates.remove(key);`?

Reply:
It isn't removed while deactivated. Here are the important snippets:

```java
// for removed policies, we add them back
          adjChildPolicies.put(entry.getKey(), entry.getValue());
// and then repackage the config
      config = new ClusterManagerConfig(adjChildPolicies);
      resolvedAddresses =
          resolvedAddresses.toBuilder().setLoadBalancingPolicyConfig(config).build();
```

The model is ClusterManagerLb tells MultiChildLb _all_ the children it is interested in. And MultiChildLb manages those. ClusterManagerLb managers the list of children it is interested in.